### PR TITLE
Equivalency between function calls, abspath(p1, p2..) and abspath(joinpath(p1, p2..))

### PR DESCRIPTION
**Pull request by sudk1896**
This PR solves #18406.
## What does this PR do ?

This PR is a doc-fix. As mentioned in #18406, this PR documents the fact that the two function calls `abspath(p1, p2...)` and `abspath(joinpath(p1, p2...))` are equivalent.

@stevengj: Kindly review. The docs compile properly on my local system.

### CODE REVIEW

File: doc/stdlib/file.rst [hunk 1]
```diff
@@ -412,6 +412,9 @@
 
    Convert a path to an absolute path by adding the current directory if necessary.
 
+   .. note::
+       The function calls ``abspath(p1, p2...)`` and ``abspath(joinpath(p1, p2...))`` are equivalent.
+
```
Comment (yuyichao) [inline doc/stdlib/file.rst]: The doc needs to be changed in the docstring `base/` and sync here.

Comment (sudk1896) [inline doc/stdlib/file.rst]: @yuyichao: My bad, I had commented asking the same (whether docstrings need to be changed in source or in the .rst files in the original issue). Didn't receive a reply so went ahead. Does that mean, I have to change the docstrings in both the ```.rst``` files and the source.
Thanks.

Comment (yuyichao) [inline doc/stdlib/file.rst]: If you change the docstring and run `doc/genstdlib.jl` with the newly compiled julia it should sync the doc here automatically.

**Comment by sudk1896** [comment]
@yuyichao: It isn't happening. I created a new branch and did.
1. Edit the docstring in /base/path.jl
2. then ran ./julia doc/genstdlib.jl
The ```.rst``` (I checked the ```file.rst``` file specifically which documents ```abspath```) files do not change one bit. Any idea why this is happening.

**Comment by yuyichao** [comment]
You need to run

> with the newly compiled julia

i.e. you need to compile julia

**Comment by sudk1896** [comment]
@yuyichao: Didn't know the new part meant, running ```make``` again. It works fine now.

**Comment by sudk1896** [comment]
@yuyichao @stevengj : I edited the doc-string in the source as asked. The html files compile fine on my machine. Review.

**Comment by tkelman** [comment]
these might be better to document as two separate docstrings for the single arg vs vararg signature

**Comment by tkelman** [comment]
closed by https://github.com/JuliaLang/julia/issues/21626

**Pull request closed without merging**